Content.Server/Storage/Components/SpawnTableOnUseComponent.cs

```diff
@@ -0,0 +1,24 @@
+using Content.Server.Storage.EntitySystems;
+using Content.Shared.EntityTable.EntitySelectors;
+using Robust.Shared.Audio;
+
+namespace Content.Server.Storage.Components;
+
+/// <summary>
+/// Spawns items from an entity table when used in hand.
+/// </summary>
+[RegisterComponent, Access(typeof(SpawnTableOnUseSystem))]
+public sealed partial class SpawnTableOnUseComponent : Component
+{
+    /// <summary>
+    /// The entity table to select entities from.
+    /// </summary>
+    [DataField(required: true)]
+    public EntityTableSelector Table = default!;
+
+    /// <summary>
+    /// A sound to play when the items are spawned. For example, gift boxes being unwrapped.
+    /// </summary>
+    [DataField]
+    public SoundSpecifier? Sound;
```

Comment:
`EmitSoundOnUse` already exists, so it seems to me like this component should only contain `Table`, and the system only be concerned with the spawning logic.

Reply:
true

Reply:
actually not true since EmitSoundOnUse plays it attached to the source, this uses the coords since it is deleted after